providers.bzl

```diff
@@ -1,90 +1,48 @@
-"""Providers for interop between JS rules.
-
-This file has to live in the built-in so that all rules can load() the providers
-even if users haven't installed any of the packages/*
-
-These providers allows rules to interoperate without knowledge
-of each other.
-
-You can think of a provider as a message bus.
-A rule "publishes" a message as an instance of the provider, and some other rule
-subscribes to these by having a (possibly transitive) dependency on the publisher.
-
-## Debugging
-
-Debug output is considered orthogonal to these providers.
-Any output may or may not have user debugging affordances provided, such as
-readable minification.
-We expect that rules will have a boolean `debug` attribute, and/or accept the `DEBUG`
-environment variable.
-Note that this means a given build either produces debug or non-debug output.
-If users really need to produce both in a single build, they'll need two rules with
-differing 'debug' attributes.
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Public providers, aspects and helper surface is re-exported here.
+
+Users should not load files under "/providers"
```

Comment:
if we really don't want users loading them, I think they should be `/internal/providers/*.bzl`

Reply:
I pushed the commit for this since you're on the road